### PR TITLE
Add more zeus modules

### PR DESCRIPTION
**Pull request by SilentSpike**
**When merged this pull request will:**
- Add a module to globally edit AI
- Add a module to teleport players
- Update our vanilla function overwrites

**Todo:**
- [x] Fix cargo vehicle attribute not displaying
- [x] Possibly rewrite the AI skill module (not sure if ace setttings are suitable here)
- [x] Add stringtable entries and function headers
- [ ] Full heal medical module
- [ ] Damage units module
- [x] AI command modules (garrison/search/patrol/etc.)
- [x] Change group side module

Please make requests/suggestions for more useful zeus modules on this PR

### CODE REVIEW

**Comment by Cuel** [comment]
Any chance to see a function like Ares_fnc_registerCustomModule?

**Comment by SilentSpike** [comment]
What does it do? Plus, should probably get in touch with the author of Ares if I'm going to be implementing overlapping modules :smile:

**Comment by Cuel** [comment]
It allows you to add custom Zeus stuff, e.g

```
["Menu root", "Sub cateogry", {_this select 0 = pos, _this select 1 = object under cursor}] call Ares_fnc_RegisterCustomModule;
```

**Comment by SilentSpike** [comment]
I could potentially add a module to inject reusable code "modules" into the zeus tree view, but that's pretty hacky so I'm not sure if we'd want to.

**Comment by Cuel** [comment]
Does it have to be hacky tho :hammer:

**Comment by SilentSpike** [comment]
It's inherently hacky :laughing:

**Comment by SilentSpike** [comment]
The AI modules are functional, but will function better once https://github.com/CBATeam/CBA_A3/pull/333 hopefully gets pulled

**Comment by TheGrandVizier** [comment]
Question here regarding AI abilities like "garrison".
Currently there is a similar modules from the abandoned ARES mod. The problem with it is that it does not work if the target AI is on HC.

Will your take on this module be HC-proof, theoretically?

**Comment by SilentSpike** [comment]
Locality shouldn't be an issue no :smile:

**Comment by commy2** [comment]
- [ ] CBA events

File: addons/zeus/functions/fnc_ui_defendArea.sqf [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    if (isNull _display) exitWith {};
+
+    private _logic = GETMVAR(BIS_fnc_initCuratorAttributes_target,objnull);
+    if (isNull _logic) exitWith {};
+
+    private _unit = effectiveCommander (attachedTo _logic);
+    private _radius = GETVAR(_display,GVAR(radius),50);
+    private _position = GETVAR(_display,GVAR(position),getPos _logic);
+
+    [QGVAR(moduleDefendArea), _unit, [_unit,_position,_radius]] call EFUNC(common,targetEvent);
```
Comment (Glowbal) [inline addons/zeus/functions/fnc_ui_defendArea.sqf]: Using old event framework

File: addons/zeus/functions/fnc_ui_patrolArea.sqf [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    if (isNull _display) exitWith {};
+
+    private _logic = GETMVAR(BIS_fnc_initCuratorAttributes_target,objnull);
+    if (isNull _logic) exitWith {};
+
+    private _unit = effectiveCommander (attachedTo _logic);
+    private _radius = GETVAR(_display,GVAR(radius),50);
+    private _position = GETVAR(_display,GVAR(position),getPos _logic);
+
+    [QGVAR(modulePatrolArea), _unit, [_unit,_position,_radius,5]] call EFUNC(common,targetEvent);
```
Comment (Glowbal) [inline addons/zeus/functions/fnc_ui_patrolArea.sqf]: Using old event framework

File: addons/zeus/functions/fnc_ui_searchArea.sqf [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    private _radius = GETVAR(_display,GVAR(radius),50);
+    private _position = GETVAR(_display,GVAR(position),getPos _logic);
+    private _marker = QGVAR(ModuleSearchArea) + str(_unit);
+
+    createMarker [_marker, _position];
+    _marker setMarkerAlpha 0;
+    _marker setMarkerShape "ELLIPSE";
+    _marker setMarkerSize [_radius,_radius];
+
+    [QGVAR(moduleSearchArea), _unit, [_unit,_marker]] call EFUNC(common,targetEvent);
```
Comment (Glowbal) [inline addons/zeus/functions/fnc_ui_searchArea.sqf]: Using old event framework

**Comment by Glowbal** [comment]
We should wrap up what is in this PR now, so we can get it in for 3.6.0. All modules that are still wip or todo can be added later with either a different PR for 3.6.0 or for 3.7.0.

**Comment by SilentSpike** [comment]
Should now be good to go!

**Pull request merged**